Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -465,21 +465,18 @@ class MessageUpdater: Worker {
                 throw ClientError.MessageDoesNotExist(messageId: messageId)
             }
 
-            switch messageDTO.localMessageState {
-            case nil, .pendingSync, .syncingFailed, .deletingFailed:
-                try session.pin(message: messageDTO, pinning: pinning)
-                messageDTO.localMessageState = .pendingSync
-            case .pendingSend, .sendingFailed:
+            do {
                 try session.pin(message: messageDTO, pinning: pinning)
-                messageDTO.localMessageState = .pendingSend
-            case .sending, .syncing, .deleting:
-                throw ClientError.MessageEditing(
-                    messageId: messageId,
-                    reason: "message is in `\(messageDTO.localMessageState!)` state"
-                )
+            } catch {
+                log.warning("Failed to optimistically add the message pinning to the database: \(error)")
             }
-        }, completion: {
-            completion?($0)
+        }, completion: { error in
+            let endpoint: Endpoint<EmptyResponse> = .pinMessage(messageId: messageId, request: .init(set: .init(pinned: true)))
+            self.apiClient.request(endpoint: endpoint) { result in
+                guard let error = result.error else { return }
+                completion?(error)
```

Comment:
So here, we probably should unpin if it fails

Reply:
yes, that makes sense.

Reply:
Updated 👍